strats.md

```diff
@@ -1075,6 +1084,55 @@ There are additional requirements depending on the exit condition:
+- _movementType_: This takes one of three possible values, "controlled", "uncontrolled", and "any", indicating the type of bounce that is required.
+  - "controlled" refers to movement type $12, which occurs when jumping using Spring Ball while rolling on the ground (e.g. from a mockball). In this state it is possible to control the height of each bounce by releasing jump.
+  - "uncontrolled" refers to movement type $13, which occurs when performing a lateral mid-air morph high enough that the morph animation completes before landing and bouncing. This state also occurs when rolling off of a ledge (e.g. after a mockball). In this state, Samus' horizontal speed will reach a slightly higher value, and without the need for a longer landing platform to accelerate on. However, it will not be possible to control the height of the bounce.
+
+A `comeInWithBlueSpringBallBounce` entrance condition must match with a `leaveWithSpringBallBounce`, `leaveWithMockball`, or `leaveWithRunway` on the other side of the door. The following requirement applies in every case:
+- The `canSpringBallBounce` tech (including `Morph` and `SpringBall` items) is implicitly required.
+
+There are additional requirements depending on the exit condition:
+
+- A match with `leaveWithBlueSpringBallBounce` has the following additional requirements:
+  - The `movementType` of the exit condition must equal that of the entrance condition, or one of them must be "any".
+  - If the `movementType` of either the exit condition or entrance condition is "controlled", then `canSpeedball` is required.
+  - `remoteAndLandingMinTiles` must contain at least one pair `(minRemoteLength, minLandingLength)` such that the effective length of the `remoteRunway` is at least `minRemoteLength` and the effective length of the `landingRunway` is at least `minLandingLength`.
+  - The `blue` property of the exit condition must be "yes" or "any". 
+  - A `getBlueSpeed` requirement based on remote runway length.
+- A match with `leaveWithMockball` has the following requirements:
+  - The `movementType` of the entrance condition must be "controlled" or "any".
+  - The `canMockball` tech is required.
+  - `remoteAndLandingMinTiles` must contain at least one pair `(minRemoteLength, minLandingLength)` such that the effective length of the `remoteRunway` is at least `minRemoteLength` and the effective length of the `landingRunway` is at least `minLandingLength`.
+  - The `blue` property of the exit condition must be "yes" or "any".
+  - A `getBlueSpeed` requirement based on remote runway length.
+- A match with `leaveWithRunway` has the following requirements:
+  - The effective runway length must be at least `adjacentMinTiles`.
+  - If the previous room is heated, then `heatFrames` are included based on the time spent running in that room. The minimally required heat frames are calculated the same way as for `comeInRunning` (and `comeInJumping`).
```

Comment:
Yeah, I think a `minTiles` on the exiting side (to ensure being able to reach the door) would need to be matched with something like a `maxTiles` on the entrance side (to ensure low enough speed to be able to adequately control the bounces). The thing I've been struggling with is how to capture the relationship between tiles, momentum values, and shortcharge requirements when getting blue speed.

When using a runway purely for momentum, you hold dash for the whole segment of runway, and there's a straightforward conversion between tiles of runway used and the final horizontal speed attained.

When using a runway for a combination of gaining blue speed and gaining momentum, it's more complicated: you're only holding dash part of the time, and even over the same length of runway the final momentum can vary depending on how the shortcharge is executed. To minimize final momentum, you would do a multi-stutter and a 4-tap with as close to perfect taps as possible. To maximize final momentum (assuming a fixed-length runway of, say, less than 27 tiles), I think you still want a multi-stutter and perfect taps but as few taps as necessary, pressing and holding dash for the final stretch of the run, ideally timed so that you get blue just at the end of the runway. So on both the low and high end, the range of momentum values that a player can attain while getting blue on a fixed length of runway depends on shortcharge ability. So this creates a problem that the difficulty level of the shortcharge can no longer be measured by just the length of runway, because it depends on the required momentum too (e.g. on how close the required momentum is to either the low or high extreme of what is possible). We could try to translate the shortcharge difficulty into a standard shortcharge runway length, based on the crispness of taps/stutters required. But I'm fairly fuzzy on exactly what this should look like or if this is the right approach. 

The tech "canSlowShortCharge" was created to solve this on the low-momentum side, but for what we're trying to do now it seems like we need a more granular solution. For these kinds of strats, I'm wondering if it might be better to encode constraints on momentum directly instead of using runway length as a proxy. When we get to setups and applications in heated rooms, the momentum constraints could also have a significant impact on heat frames.